third_party/terraform/resources/resource_google_project_service.go

```diff
@@ -187,9 +187,21 @@ func resourceGoogleProjectServiceRead(d *schema.ResourceData, meta interface{})
 		return nil
 	}
 
-	// The service is was not found in enabled services - remove it from state
-	log.Printf("[DEBUG] service %s not in enabled services for project %s, removing from state", srv, project)
-	d.SetId("")
+	// only remove the project from state when it's not found if this isn't
+	// a Read that was called through from Create. The call through from
+	// Create may be dealing with eventual consistency, and will try to
+	// remove the newly created resource from state, which doesn't match
+	// the plan. So the user will get an incomprehensible error about
+	// "produced an unexpected new value for was present, but now absent."
+	// if we don't have this check.
+	//
+	// If the service _actually_ didn't get enabled, it'll be removed by
```

Comment:
This seems like a degradation in behaviour- should we return an error if we're in create or something?

Reply:
@paddycarver has a bit more context around this as well. If we decide to keep it, I'll pull it out in a separate PR - it was an eventual consistency error that was masking a different error that was fixed in the SDK.

Reply:
There are a few ways this can be handled. You can retry a few times and see if it shows up properly once eventual consistency has settled down. You can return an error. You can _probably_ just ignore it, like we do here.

Removing this though risks a "plan did not match apply" error because you planned to have a project service resource, and then didn't set one in state, so downstream resources will get undefined inputs if they interpolate off this.

Reply:
We can both `d.SetId("")` and return an error, right? That way we'll have no resource in state + return an error, halting the rest of the apply. That seems reasonable.

Reply:
You may be able to?

In theory, the state shouldn't be updated at all if you return an error, but there's an existing SDK bug around that so that's out the window. I'm not sure if core will complain or not, honestly.

I think if you return the error, you may want to _not_ set the ID, as 1) it may be in error due to eventual consistency, and if that's the case the next apply will fix things instead of re-creating the resource and 2) if it's not in error, the next reply will (correctly) remove it from state, anyways. Seeing as no downstreams can possibly get the result, as we're halting execution, it's safe to rely on refresh for putting the state back in order.